pkg/providers/cloudstack/validator.go

```diff
@@ -148,17 +168,32 @@ func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, cloudStac
 }
 
 func (v *Validator) validateMachineConfig(ctx context.Context, datacenterConfigSpec anywherev1.CloudStackDatacenterConfigSpec, machineConfig *anywherev1.CloudStackMachineConfig) error {
-	domain := datacenterConfigSpec.Domain
-	zone := datacenterConfigSpec.Zone
-	account := datacenterConfigSpec.Account
-	if err := v.cmk.ValidateTemplatePresent(ctx, domain, zone, account, machineConfig.Spec.Template); err != nil {
-		return fmt.Errorf("validating template: %v", err)
+	if len(datacenterConfigSpec.Domain) <= 0 {
+		return fmt.Errorf("CloudStackDatacenterConfig domain is not set or is empty")
```

Comment:
If there's no formatting to be done on the string just use `errors.New()`. `fmt.Errorf()` invokes a bunch of formatting logic.